### PR TITLE
Sam/dev-firebase-error

### PR DESCRIPTION
**Pull request by samholmes**
### CHANGELOG

 none

### Dependencies

 none

### Requirements

If you have made **any** visual changes to the GUI. Make sure you have:

- [ ] Tested on iOS device
- [ ] Tested on Android device
- [ ] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203768363890533

### CODE REVIEW

File: src/components/scenes/LoginScene.tsx [hunk 1]
```diff
@@ -82,7 +82,7 @@ class LoginSceneComponent extends React.PureComponent<Props, State> {
       // @ts-expect-error
       .catch(e => this.setState({ backgroundImage: theme.backgroundImage }))
     const { YOLO_USERNAME, YOLO_PASSWORD, YOLO_PIN } = ENV
-    if (YOLO_USERNAME != null && (YOLO_PASSWORD != null || YOLO_PIN != null) && firstRun) {
+    if (YOLO_USERNAME != null && (Boolean(YOLO_PASSWORD) || Boolean(YOLO_PIN)) && firstRun) {
```
Comment (swansontec) [inline src/components/scenes/LoginScene.tsx:85]: Why not get the username too? Also, this is a pretty cool pattern we might use in more places.

**Review by swansontec — APPROVED** [review]
Approved, but with question.

File: src/components/scenes/LoginScene.tsx (hunk 1 above)
Comment (samholmes) [reply src/components/scenes/LoginScene.tsx:85]: Fair point; Username could also be given this treatment. The type-casting pattern is nice, and I'm happy to promote it.

**Review by swansontec — APPROVED** [review]

File: src/components/scenes/LoginScene.tsx [hunk 2]
```diff
@@ -82,7 +82,7 @@ class LoginSceneComponent extends React.PureComponent<Props, State> {
       // @ts-expect-error
       .catch(e => this.setState({ backgroundImage: theme.backgroundImage }))
     const { YOLO_USERNAME, YOLO_PASSWORD, YOLO_PIN } = ENV
-    if (YOLO_USERNAME != null && (YOLO_PASSWORD != null || YOLO_PIN != null) && firstRun) {
+    if (!!YOLO_USERNAME && (!!YOLO_PASSWORD || !!YOLO_PIN) && firstRun) {
```
Comment (swansontec) [inline src/components/scenes/LoginScene.tsx:85]: Not approved. Do the `Boolean()` thing instead. We eventually want to enable a lint rule that disallows treating strings and other data types as booleans, such as using `!` or `if` or `?`. The `Boolean()` function encapsulates the `!= null && !== ''` logic in a way ESLint will be happy with.

**Review by swansontec — CHANGES_REQUESTED** [review]
Oh, good call on finally fixing up env.json

**Review by swansontec — APPROVED** [review]

**Pull request merged**